index.js

```diff
@@ -118,6 +127,7 @@ class MultiStream extends stream.Readable {
       stream.removeListener('readable', onReadable)
       stream.removeListener('end', onEnd)
       stream.removeListener('close', onClose)
+      stream.destroy()
```

Comment:
not all streams will auto destroy on 'end', so we should explicitly destroy just to make sure we don't have any dangling resources in buggy streams